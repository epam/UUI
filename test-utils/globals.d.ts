declare module "jsurl";
declare module "query-string";
