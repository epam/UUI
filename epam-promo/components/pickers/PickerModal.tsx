import * as React from 'react';
import css from './PickerModal.scss';
import { DataRowProps, Lens } from '@epam/uui-core';
import { PickerModalBase, PickerModalProps, handleDataSourceKeyboard, IconContainer } from '@epam/uui-components';
import { DataPickerBody } from './DataPickerBody';
import { FlexRow, FlexCell, FlexSpacer } from '../layout/FlexItems';
import { ModalBlocker, ModalWindow, ModalHeader, ModalFooter } from '../overlays';
import { SearchInput, Switch } from '../inputs';
import { LinkButton, Button } from '../buttons';
import { DataPickerRow } from './DataPickerRow';
import { Text, TextPlaceholder } from '../typography';
import { i18n } from '../../i18n';
import { ReactComponent as SearchIcon } from '../../icons/search-with-background.svg';

export class PickerModalImpl<TItem, TId> extends PickerModalBase<TItem, TId> {
    renderRow(rowProps: DataRowProps<TItem, TId>) {
        return this.props.renderRow ? (
            this.props.renderRow(rowProps, this.state.dataSourceState)
        ) : (
            <DataPickerRow
                {...rowProps}
                key={rowProps.rowKey}
                borderBottom="none"
                padding="24"
                size="36"
                renderItem={i => <Text size="36">{rowProps.isLoading ? <TextPlaceholder wordsCount={2} /> : this.getName(i)}</Text>}
            />
        );
    }

    renderFooter(selectedDataRows: DataRowProps<TItem, TId>[]) {
        const hasSelection = selectedDataRows.length > 0;
        const view = this.getView();

        return (
            <>
                {view.selectAll && (
                    <LinkButton
                        caption={hasSelection ? i18n.pickerModal.clearAllButton : i18n.pickerModal.selectAllButton}
                        onClick={hasSelection ? () => this.clearSelection() : () => view.selectAll.onValueChange(true)}
                    />
                )}
                <FlexSpacer />
                <Button fill="white" color="gray50" caption={i18n.pickerModal.cancelButton} onClick={() => this.props.abort()} />
                <Button color="green" caption={i18n.pickerModal.selectButton} onClick={() => this.props.success(null)} />
            </>
        );
    }

    renderNotFound = () => {
        return this.props.renderNotFound ? (
            this.props.renderNotFound({ search: this.state.dataSourceState.search, onClose: () => this.props.success(null) })
        ) : (
            <div className={css.noFoundModalContainer}>
                <IconContainer cx={css.noFoundModalContainerIcon} icon={SearchIcon} />
                <Text cx={css.noFoundModalContainerText} font="sans-semibold" fontSize="16" lineHeight="24" color="gray80" size={'36'}>
                    {i18n.dataPickerBody.noRecordsMessage}
                </Text>
                <Text cx={css.noFoundModalContainerText} fontSize="12" lineHeight="18" font="sans" color="gray80" size={'36'}>
                    {i18n.dataPickerBody.noRecordsSubTitle}
                </Text>
            </div>
        );
    };

    render() {
        const view = this.getView();
        const dataRows = this.getRows();
        const selectedDataRows = view.getSelectedRows();
        const rows = dataRows.map(props => this.renderRow(props));

        return (
            <ModalBlocker blockerShadow="dark" {...this.props}>
                <ModalWindow width="600" height="700">
                    <ModalHeader borderBottom title={this.props.caption || i18n.pickerModal.headerTitle} onClose={() => this.props.abort()} />
                    <FlexCell cx={css.subHeaderWrapper}>
                        <FlexRow vPadding="24">
                            <SearchInput
                                {...this.lens.prop('dataSourceState').prop('search').toProps()}
                                onKeyDown={e =>
                                    handleDataSourceKeyboard(
                                        {
                                            value: this.getDataSourceState(),
                                            onValueChange: this.handleDataSourceValueChange,
                                            listView: view,
                                            rows: dataRows,
                                            editMode: 'modal',
                                        },
                                        e
                                    )
                                }
                                autoFocus={true}
                                placeholder={i18n.pickerModal.searchPlaceholder}
                            />
                        </FlexRow>
                        {!this.isSingleSelect() && (
                            <Switch
                                cx={css.switch}
                                size="18"
                                {...this.stateLens.prop('showSelected').toProps()}
                                isDisabled={selectedDataRows.length < 1}
                                label="Show only selected"
                            />
                        )}
                        {this.props.renderFilter && (
                            <FlexCell grow={2}>{this.props.renderFilter(this.lens.prop('dataSourceState').prop('filter').toProps())}</FlexCell>
                        )}
                    </FlexCell>
                    <DataPickerBody
                        {...this.getListProps()}
                        value={this.getDataSourceState()}
                        onValueChange={this.handleDataSourceValueChange}
                        search={this.lens.prop('dataSourceState').prop('search').toProps()}
                        showSearch={false}
                        rows={rows}
                        renderNotFound={this.renderNotFound}
                        editMode="modal"
                    />
                    <ModalFooter borderTop padding="24" vPadding="24">
                        {this.props.renderFooter ? this.props.renderFooter(this.getFooterProps()) : this.renderFooter(selectedDataRows)}
                    </ModalFooter>
                </ModalWindow>
            </ModalBlocker>
        );
    }
}

export class PickerModal<TItem, TId> extends React.Component<PickerModalProps<TItem, TId>> {
    state = { selection: this.props.initialValue };
    lens = Lens.onState<any>(this);
    render() {
        return (
            <PickerModalImpl
                {...this.props}
                {...this.lens.prop('selection').toProps()}
                success={() => this.props.success(this.state.selection as any)}
            />
        );
    }
}
