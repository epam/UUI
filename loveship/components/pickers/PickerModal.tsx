import * as React from 'react';
import { DataRowProps, Lens } from '@epam/uui-core';
import { PickerModalBase, PickerModalProps, handleDataSourceKeyboard } from '@epam/uui-components';
import { DataPickerBody } from './DataPickerBody';
import { FlexRow, FlexCell, FlexSpacer } from '../layout/FlexItems';
import { ModalBlocker, ModalWindow, ModalHeader } from '../overlays';
import { SearchInput, Switch } from '../inputs';
import { LinkButton, Button } from '../buttons';
import { DataPickerRow } from './DataPickerRow';
import { Text, TextPlaceholder } from '../typography';
import { i18n } from '../../i18n';
import css from './PickerModal.scss';

export class PickerModalImpl<TItem, TId> extends PickerModalBase<TItem, TId> {
    renderRow(rowProps: DataRowProps<TItem, TId>) {
        return this.props.renderRow ? (
            this.props.renderRow(rowProps, this.state.dataSourceState)
        ) : (
            <DataPickerRow
                {...rowProps}
                key={rowProps.rowKey}
                borderBottom="none"
                padding="24"
                size="36"
                renderItem={i => (
                    <Text size="36" color={rowProps.isDisabled ? 'night500' : 'night700'}>
                        {rowProps.isLoading ? <TextPlaceholder wordsCount={2} /> : this.getName(i)}
                    </Text>
                )}
            />
        );
    }

    renderFooter(selectedDataRows: DataRowProps<TItem, TId>[]) {
        const hasSelection = selectedDataRows.length > 0;
        const view = this.getView();

        return (
            <>
                {view.selectAll && (
                    <LinkButton
                        caption={hasSelection ? i18n.pickerModal.clearSelectionButton : i18n.pickerModal.selectAllButton}
                        onClick={hasSelection ? () => this.clearSelection() : () => view.selectAll.onValueChange(true)}
                    />
                )}
                <FlexSpacer />
                <Button fill="white" color="night500" caption={i18n.pickerModal.cancelButton} onClick={() => this.props.abort()} />
                <Button color="grass" caption={i18n.pickerModal.selectButton} onClick={() => this.props.success(null)} />
            </>
        );
    }

    render(): React.ReactNode {
        const view = this.getView();
        const dataRows = this.getRows();
        const selectedDataRows = view.getSelectedRows();
        const rows = dataRows.map(props => this.renderRow(props));

        return (
            <ModalBlocker blockerShadow="dark" {...this.props}>
                <ModalWindow width="600" height="700">
                    <ModalHeader borderBottom title={this.props.caption || i18n.pickerModal.headerTitle} onClose={() => this.props.abort()} />
                    <FlexCell cx={css.subHeaderWrapper}>
                        <FlexRow vPadding="24">
                            <SearchInput
                                {...this.lens.prop('dataSourceState').prop('search').toProps()}
                                onKeyDown={e =>
                                    handleDataSourceKeyboard(
                                        {
                                            value: this.getDataSourceState(),
                                            onValueChange: this.handleDataSourceValueChange,
                                            listView: view,
                                            rows: dataRows,
                                            editMode: 'modal',
                                        },
                                        e
                                    )
                                }
                                autoFocus={true}
                                placeholder={i18n.pickerModal.searchPlaceholder}
                            />
                        </FlexRow>
                        {!this.isSingleSelect() && (
                            <Switch
                                cx={css.switch}
                                color="sky"
                                size="18"
                                value={this.state.showSelected}
                                onValueChange={nV =>
                                    this.setState({ showSelected: nV, dataSourceState: { ...this.state.dataSourceState, search: '' } })
                                }
                                isDisabled={selectedDataRows.length < 1}
                                label="Show only selected"
                            />
                        )}
                        {this.props.renderFilter && (
                            <FlexCell grow={2}>{this.props.renderFilter(this.lens.prop('dataSourceState').prop('filter').toProps())}</FlexCell>
                        )}
                    </FlexCell>
                    <DataPickerBody
                        {...this.getListProps()}
                        value={this.getDataSourceState()}
                        onValueChange={this.handleDataSourceValueChange}
                        search={this.lens.prop('dataSourceState').prop('search').toProps()}
                        showSearch={false}
                        rows={rows}
                        renderNotFound={
                            this.props.renderNotFound &&
                            (() =>
                                this.props.renderNotFound({
                                    search: this.state.dataSourceState.search,
                                    onClose: () => this.props.success(selectedDataRows),
                                }))
                        }
                        editMode="modal"
                    />
                    <FlexRow padding="24" vPadding="24" spacing="12" cx={css.modalFooter}>
                        {this.props.renderFooter ? this.props.renderFooter(this.getFooterProps()) : this.renderFooter(selectedDataRows)}
                    </FlexRow>
                </ModalWindow>
            </ModalBlocker>
        );
    }
}

export class PickerModal<TItem, TId> extends React.Component<PickerModalProps<TItem, TId>> {
    state = { selection: this.props.initialValue };
    lens = Lens.onState<any>(this);

    render() {
        return (
            <PickerModalImpl
                {...this.props}
                {...this.lens.prop('selection').toProps()}
                success={() => this.props.success(this.state.selection as any)}
            />
        );
    }
}
