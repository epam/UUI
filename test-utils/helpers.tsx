import React, { ComponentType, ReactElement, ReactNode } from 'react';
import { mount } from 'enzyme';
import { renderHook } from '@testing-library/react-hooks';
import { act as reactAct } from 'react-dom/test-utils';
import renderer, { act as rendererAct } from 'react-test-renderer';
import { ContextProvider, UuiContexts } from '@epam/uui-core';

export const delay = (ms: number = 1): Promise<void> =>
    new Promise(resolve => {
        setTimeout(resolve, ms);
    });

export const testSvc = {} as UuiContexts;

export async function mountHookWithContext<TProps, TResult>(hook: (props: TProps) => TResult, initialProps?: TProps) {
    const wrapper = ({ children }: { children?: React.ReactNode }) => (
        <ContextProvider onInitCompleted={svc => Object.assign(testSvc, svc)}>{children}</ContextProvider>
    );

    const { waitForNextUpdate, rerender, ...rest } = renderHook<TProps, TResult>(hook, { wrapper, initialProps: initialProps });

    await waitForNextUpdate();

    return {
        rerender: (props: TProps) => rerender({ ...props, children: undefined }),
        waitForNextUpdate,
        ...rest,
    };
}

export const mountWithContextAsync = async (children: ReactNode, enableLegacyContext = false) => {
    const wrapper = mount(
        <ContextProvider
            onInitCompleted={svc => {
                Object.assign(testSvc, svc);
            }}
            enableLegacyContext={enableLegacyContext}
        >
            {children}
        </ContextProvider>
    );

    await reactAct(delay);
    wrapper.update();

    return wrapper;
};

export const mountWrappedComponentAsync = async <TComponent extends ComponentType<TProps>, TProps>(
    Component: TComponent,
    props: TProps,
    enableLegacyContext = false
) => {
    const renderComponent = (properties: TProps) =>
        mount(
            React.createElement(
                (p: any) => (
                    <ContextProvider
                        onInitCompleted={svc => {
                            Object.assign(testSvc, svc);
                        }}
                        enableLegacyContext={enableLegacyContext}
                    >
                        <Component {...p} />
                    </ContextProvider>
                ),
                properties
            )
        );
    const wrapper = renderComponent(props);

    await reactAct(delay);
    wrapper.update();

    return wrapper;
};

export const renderWithContextAsync = async (component: ReactElement, enableLegacyContext = false) => {
    const result = renderer.create(
        <ContextProvider
            onInitCompleted={svc => {
                Object.assign(testSvc, svc);
            }}
            enableLegacyContext={enableLegacyContext}
        >
            {component}
        </ContextProvider>
    );

    await rendererAct(delay);
    return result.toJSON();
};
//
// global = {
//     ...global,
//     delay,
//     testSvc,
//     mountWithContextAsync,
//     mountWrappedComponentAsync,
//     renderWithContextAsync,
// } as any;

// // @ts-ignore
// global.delay = delay;
// // @ts-ignore
// global.testSvc = testSvc;
// // @ts-ignore
// global.mountWithContextAsync = mountWithContextAsync;
// // @ts-ignore
// global.mountWrappedComponentAsync = mountWrappedComponentAsync;
// // @ts-ignore
// global.renderWithContextAsync = renderWithContextAsync;
