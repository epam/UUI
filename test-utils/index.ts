export * from './helpers';
export * from './mocks';
