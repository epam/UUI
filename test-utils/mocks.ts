export const windowMock = {
    matchMedia: () => ({matches: false}),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    document: {
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
    },
} as any;