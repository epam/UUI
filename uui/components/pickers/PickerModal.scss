@use  '../../assets/styles/index' as *;

.sub-header-wrapper {
    border-bottom: 1px solid var(--uui-divider);
    padding: 6px 24px;
}

.switch {
    margin-bottom: 6px;
}
