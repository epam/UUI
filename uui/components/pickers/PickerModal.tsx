import React, { useMemo } from 'react';
import { IHasCaption, PickerBaseOptions, PickerRenderRowParams } from '@epam/uui-core';
import { IconContainer, PickerModalArrayProps, PickerModalOptions, PickerModalScalarProps, handleDataSourceKeyboard, usePickerModal } from '@epam/uui-components';
import { DataPickerRow } from './DataPickerRow';
import { Text } from '../typography';
import { FlexRow, FlexCell, FlexSpacer } from '../layout';
import {
    ModalBlocker, ModalWindow, ModalHeader, ModalFooter,
} from '../overlays';
import { LinkButton, Button } from '../buttons';
import { SearchInput, Switch } from '../inputs';
import { DataPickerBody } from './DataPickerBody';
import { i18n } from '../../i18n';
import { settings } from '../../settings';

import css from './PickerModal.module.scss';

export type PickerModalProps<TItem, TId> = PickerBaseOptions<TItem, TId> &
IHasCaption &
(PickerModalScalarProps<TId, TItem> | PickerModalArrayProps<TId, TItem>) &
PickerModalOptions<TItem, TId>;

export function PickerModal<TItem, TId>(props: PickerModalProps<TItem, TId>) {
    const {
        view,
        selection,
        dataSourceStateLens,
        dataSourceState,
        getName,
        clearSelection,
        getRows,
        getListProps,
        getFooterProps,
        isSingleSelect,
        handleDataSourceValueChange,
    } = usePickerModal<TItem, TId>(props);
    const isSearching = dataSourceState.search && dataSourceState.search.length > 0;

    const renderRow = (rowProps: PickerRenderRowParams<TItem, TId>) => {
        return props.renderRow ? (
            props.renderRow(rowProps, dataSourceState)
        ) : (
            <DataPickerRow
                { ...rowProps }
                key={ rowProps.rowKey }
                size={ settings.pickerInput.sizes.body.row }
                flattenSearchResults={ view.getConfig().flattenSearchResults }
                dataSourceState={ dataSourceState }
                getName={ getName }
            />
        );
    };

    const renderFooter = () => {
        const hasSelection = view.getSelectedRowsCount() > 0;
        const rowsCount = view.getListProps().rowsCount;
        const isEmptyRowsAndHasNoSelection = (rowsCount === 0 || !hasSelection);
        const showClear = !props.disableClear && (isSingleSelect() ? true : (!view.selectAll || hasSelection));
        const isClearDisabled = isSearching || isEmptyRowsAndHasNoSelection;

        return (
            <>
                {view.selectAll && !hasSelection && (
                    <LinkButton
                        caption={ i18n.pickerModal.selectAllButton }
                        onClick={ () => view.selectAll.onValueChange(true) }
                    />
                )}
                {showClear && (
                    <LinkButton
                        caption={ isSingleSelect() ? i18n.pickerModal.clearButton : i18n.pickerModal.clearAllButton }
                        onClick={ () => clearSelection() }
                        isDisabled={ isClearDisabled }
                    />
                )}
                <FlexSpacer />
                <Button fill="outline" color="secondary" caption={ i18n.pickerModal.cancelButton } onClick={ () => props.abort() } />
                <Button color="primary" caption={ i18n.pickerModal.selectButton } onClick={ () => props.success(selection as any) } />
            </>
        );
    };

    const renderNotFound = () => {
        return props.renderNotFound ? (
            props.renderNotFound({ search: dataSourceState.search, onClose: () => props.success(null) })
        ) : (
            <div className={ css.noFoundModalContainer }>
                <IconContainer cx={ css.noFoundModalContainerIcon } icon={ settings.pickerInput.icons.body.modalNotFoundSearchIcon } />
                <Text cx={ css.noFoundModalContainerText } fontWeight="600" fontSize="16" lineHeight="24" color="primary">
                    {i18n.dataPickerBody.noRecordsMessage}
                </Text>
                <Text cx={ css.noFoundModalContainerText } fontSize="12" lineHeight="18" color="primary">
                    {i18n.dataPickerBody.noRecordsSubTitle}
                </Text>
            </div>
        );
    };

    const dataRows = getRows();

    const focusedRowId = useMemo((): string => {
        const focusedRow = dataRows.find((row) => {
            return row.isFocused;
        });

        if (!focusedRow) {
            return '';
        }

        return focusedRow.rowKey;
    }, [dataRows]);

    return (
        <ModalBlocker { ...props }>
            <ModalWindow width={ 600 } height={ 700 } cx={ css.body }>
                <ModalHeader title={ props.caption || i18n.pickerModal.headerTitle } onClose={ () => props.abort() } />
                <FlexCell cx={ css.subHeaderWrapper }>
                    <FlexRow vPadding="24">
                        <SearchInput
                            { ...dataSourceStateLens.prop('search').toProps() }
                            onKeyDown={ (e) =>
                                handleDataSourceKeyboard(
                                    {
                                        value: dataSourceState,
                                        onValueChange: handleDataSourceValueChange,
                                        listView: view,
                                        rows: dataRows,
                                        searchPosition: 'body',
                                    },
                                    e,
                                ) }
                            autoFocus={ true }
                            placeholder={ i18n.pickerModal.searchPlaceholder }
                            rawProps={ {
                                dir: 'auto',
                                'aria-activedescendant': focusedRowId,
                            } }
                        />
                    </FlexRow>
                    {!isSingleSelect() && (
                        <Switch
                            cx={ css.switch }
                            size={ settings.pickerInput.sizes.body.footerSwitchMap[settings.pickerInput.sizes.body.row] }
                            { ...getFooterProps().showSelected }
                            isDisabled={ isSearching || view.getSelectedRowsCount() < 1 }
                            label="Show only selected"
                        />
                    )}
                    {props.renderFilter && <FlexCell grow={ 2 }>{props.renderFilter(dataSourceStateLens.prop('filter').toProps())}</FlexCell>}
                </FlexCell>
                <DataPickerBody
                    { ...getListProps() }
                    value={ dataSourceState }
                    onValueChange={ handleDataSourceValueChange }
                    showSearch={ false }
                    getName={ getName }
                    rows={ dataRows }
                    renderRow={ renderRow }
                    renderEmpty={ renderNotFound }
                    size={ settings.pickerInput.sizes.body.row }
                />
                <ModalFooter>
                    {props.renderFooter ? props.renderFooter(getFooterProps()) : renderFooter()}
                </ModalFooter>
            </ModalWindow>
        </ModalBlocker>
    );
}
