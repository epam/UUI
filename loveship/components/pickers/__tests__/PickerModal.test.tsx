import React from 'react';
import { renderWithContextAsync } from '@epam/test-utils';
import { dataSource } from './dataMocks';
import { PickerModal } from '../PickerModal';

describe('PickerModal', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderWithContextAsync(
            <PickerModal
                key="test"
                valueType="id"
                dataSource={dataSource}
                success={jest.fn()}
                abort={jest.fn()}
                zIndex={1}
                selectionMode="single"
                initialValue={null}
                isActive
            />
        );
        expect(tree).toMatchSnapshot();
    });
});
