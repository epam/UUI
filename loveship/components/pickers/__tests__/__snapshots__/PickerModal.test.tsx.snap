// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerModal should be rendered correctly 1`] = `
<div
  className="container modalBlocker shadow-dark"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="uui-modal-blocker"
    onClick={[Function]}
  />
  <div
    aria-modal={true}
    className="uui-modal-window modal width-600 height-700 container"
    role="modal"
  >
    <div
      className="root size-36 background-none padding-24 vPadding-12 border-bottom-night400 spacing-6 modalHeader container align-items-center"
    >
      <div
        className="text font-sans-semibold color-night700 line-height-24 font-size-18 v-padding-12 container"
      >
        Please select
      </div>
      <div
        className="flexSpacer"
      />
      <div
        className="flexCell"
        onClick={[Function]}
        style={
          Object {
            "alignSelf": undefined,
            "flexBasis": "auto",
            "flexGrow": undefined,
            "flexShrink": 0,
            "minWidth": 0,
            "textAlign": undefined,
          }
        }
      >
        <div
          className="root color-night600 uui-button-box uui-enabled -clickable"
          href={null}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="container uui-icon uui-enabled"
            style={Object {}}
          >
            <svg
              className=""
              height={24}
              viewBox="0 0 12 24"
              width={12}
            >
              <use
                xlinkHref="#navigation-close-24.svg"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      className="flexCell subHeaderWrapper"
      onClick={[Function]}
      style={
        Object {
          "alignSelf": undefined,
          "flexBasis": "0",
          "flexGrow": undefined,
          "flexShrink": undefined,
          "minWidth": 0,
          "textAlign": undefined,
        }
      }
    >
      <div
        className="root size-36 background-none vPadding-24 spacing-6 container align-items-center"
      >
        <div
          className="container uui-input-box -clickable colorSky root size-36 mode-form"
          onBlur={[Function]}
          onFocus={[Function]}
          tabIndex={-1}
        >
          <div
            className="container uui-icon uui-enabled"
            style={Object {}}
          >
            <svg
              className=""
              height={24}
              viewBox="0 0 12 24"
              width={12}
            >
              <use
                xlinkHref="#search-18.svg"
              />
            </svg>
          </div>
          <input
            aria-invalid={false}
            autoFocus={true}
            className="uui-input line-height-18 font-size-14 v-padding-8"
            inputMode="search"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Type text for quick search"
            type="search"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="root size-36 background-white spacing-6 container align-items-center"
      style={
        Object {
          "maxHeight": undefined,
        }
      }
    >
      <div
        className="wrapper root shadow-white"
      >
        <div
          className="body"
          style={
            Object {
              "height": "auto",
              "maxHeight": 100500,
              "minHeight": 0,
              "overflow": "hidden",
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            className="container1"
            style={
              Object {
                "WebkitOverflowScrolling": "touch",
                "bottom": undefined,
                "left": undefined,
                "marginBottom": 0,
                "marginRight": 0,
                "maxHeight": 100500,
                "minHeight": 0,
                "overflow": "scroll",
                "position": "relative",
                "right": undefined,
                "top": undefined,
              }
            }
          >
            <div
              className="container2"
              style={
                Object {
                  "minHeight": null,
                }
              }
            >
              <div
                aria-setsize={10}
                className="container3"
                role="listbox"
                style={
                  Object {
                    "marginTop": 0,
                  }
                }
              >
                <div
                  className="pickerRow -clickable container -clickable align-items-center"
                  onClick={[Function]}
                >
                  <div
                    className="flexCell cell size-36 padding-24 padding-left-24 align-widgets-top"
                    onClick={[Function]}
                    style={
                      Object {
                        "alignSelf": undefined,
                        "flexBasis": "0",
                        "flexGrow": 1,
                        "flexShrink": undefined,
                        "minWidth": 0,
                        "textAlign": undefined,
                      }
                    }
                  >
                    <div
                      className="renderItem"
                    >
                      <div
                        className="text font-sans color-night700 line-height-18 font-size-14 v-padding-9 container"
                      >
                        Ken SÃ¡nchez
                      </div>
                      <div
                        className="flexSpacer"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="pickerRow -clickable container -clickable align-items-center"
                  onClick={[Function]}
                >
                  <div
                    className="flexCell cell size-36 padding-24 padding-left-24 align-widgets-top"
                    onClick={[Function]}
                    style={
                      Object {
                        "alignSelf": undefined,
                        "flexBasis": "0",
                        "flexGrow": 1,
                        "flexShrink": undefined,
                        "minWidth": 0,
                        "textAlign": undefined,
                      }
                    }
                  >
                    <div
                      className="renderItem"
                    >
                      <div
                        className="text font-sans color-night700 line-height-18 font-size-14 v-padding-9 container"
                      >
                        Terri Duffy
                      </div>
                      <div
                        className="flexSpacer"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="pickerRow -clickable container -clickable align-items-center"
                  onClick={[Function]}
                >
                  <div
                    className="flexCell cell size-36 padding-24 padding-left-24 align-widgets-top"
                    onClick={[Function]}
                    style={
                      Object {
                        "alignSelf": undefined,
                        "flexBasis": "0",
                        "flexGrow": 1,
                        "flexShrink": undefined,
                        "minWidth": 0,
                        "textAlign": undefined,
                      }
                    }
                  >
                    <div
                      className="renderItem"
                    >
                      <div
                        className="text font-sans color-night700 line-height-18 font-size-14 v-padding-9 container"
                      >
                        Roberto Tamburello
                      </div>
                      <div
                        className="flexSpacer"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="pickerRow -clickable container -clickable align-items-center"
                  onClick={[Function]}
                >
                  <div
                    className="flexCell cell size-36 padding-24 padding-left-24 align-widgets-top"
                    onClick={[Function]}
                    style={
                      Object {
                        "alignSelf": undefined,
                        "flexBasis": "0",
                        "flexGrow": 1,
                        "flexShrink": undefined,
                        "minWidth": 0,
                        "textAlign": undefined,
                      }
                    }
                  >
                    <div
                      className="renderItem"
                    >
                      <div
                        className="text font-sans color-night700 line-height-18 font-size-14 v-padding-9 container"
                      >
                        Rob Walters
                      </div>
                      <div
                        className="flexSpacer"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="pickerRow -clickable container -clickable align-items-center"
                  onClick={[Function]}
                >
                  <div
                    className="flexCell cell size-36 padding-24 padding-left-24 align-widgets-top"
                    onClick={[Function]}
                    style={
                      Object {
                        "alignSelf": undefined,
                        "flexBasis": "0",
                        "flexGrow": 1,
                        "flexShrink": undefined,
                        "minWidth": 0,
                        "textAlign": undefined,
                      }
                    }
                  >
                    <div
                      className="renderItem"
                    >
                      <div
                        className="text font-sans color-night700 line-height-18 font-size-14 v-padding-9 container"
                      >
                        Gail Erickson
                      </div>
                      <div
                        className="flexSpacer"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="pickerRow -clickable container -clickable align-items-center"
                  onClick={[Function]}
                >
                  <div
                    className="flexCell cell size-36 padding-24 padding-left-24 align-widgets-top"
                    onClick={[Function]}
                    style={
                      Object {
                        "alignSelf": undefined,
                        "flexBasis": "0",
                        "flexGrow": 1,
                        "flexShrink": undefined,
                        "minWidth": 0,
                        "textAlign": undefined,
                      }
                    }
                  >
                    <div
                      className="renderItem"
                    >
                      <div
                        className="text font-sans color-night700 line-height-18 font-size-14 v-padding-9 container"
                      >
                        Jossef Goldberg
                      </div>
                      <div
                        className="flexSpacer"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="pickerRow -clickable container -clickable align-items-center"
                  onClick={[Function]}
                >
                  <div
                    className="flexCell cell size-36 padding-24 padding-left-24 align-widgets-top"
                    onClick={[Function]}
                    style={
                      Object {
                        "alignSelf": undefined,
                        "flexBasis": "0",
                        "flexGrow": 1,
                        "flexShrink": undefined,
                        "minWidth": 0,
                        "textAlign": undefined,
                      }
                    }
                  >
                    <div
                      className="renderItem"
                    >
                      <div
                        className="text font-sans color-night700 line-height-18 font-size-14 v-padding-9 container"
                      >
                        Dylan Miller
                      </div>
                      <div
                        className="flexSpacer"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="pickerRow -clickable container -clickable align-items-center"
                  onClick={[Function]}
                >
                  <div
                    className="flexCell cell size-36 padding-24 padding-left-24 align-widgets-top"
                    onClick={[Function]}
                    style={
                      Object {
                        "alignSelf": undefined,
                        "flexBasis": "0",
                        "flexGrow": 1,
                        "flexShrink": undefined,
                        "minWidth": 0,
                        "textAlign": undefined,
                      }
                    }
                  >
                    <div
                      className="renderItem"
                    >
                      <div
                        className="text font-sans color-night700 line-height-18 font-size-14 v-padding-9 container"
                      >
                        Diane Margheim
                      </div>
                      <div
                        className="flexSpacer"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="pickerRow -clickable container -clickable align-items-center"
                  onClick={[Function]}
                >
                  <div
                    className="flexCell cell size-36 padding-24 padding-left-24 align-widgets-top"
                    onClick={[Function]}
                    style={
                      Object {
                        "alignSelf": undefined,
                        "flexBasis": "0",
                        "flexGrow": 1,
                        "flexShrink": undefined,
                        "minWidth": 0,
                        "textAlign": undefined,
                      }
                    }
                  >
                    <div
                      className="renderItem"
                    >
                      <div
                        className="text font-sans color-night700 line-height-18 font-size-14 v-padding-9 container"
                      >
                        Gigi Matthew
                      </div>
                      <div
                        className="flexSpacer"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="pickerRow -clickable container -clickable align-items-center"
                  onClick={[Function]}
                >
                  <div
                    className="flexCell cell size-36 padding-24 padding-left-24 align-widgets-top"
                    onClick={[Function]}
                    style={
                      Object {
                        "alignSelf": undefined,
                        "flexBasis": "0",
                        "flexGrow": 1,
                        "flexShrink": undefined,
                        "minWidth": 0,
                        "textAlign": undefined,
                      }
                    }
                  >
                    <div
                      className="renderItem"
                    >
                      <div
                        className="text font-sans color-night700 line-height-18 font-size-14 v-padding-9 container"
                      >
                        Michael Raheem
                      </div>
                      <div
                        className="flexSpacer"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            style={
              Object {
                "borderRadius": 3,
                "bottom": 2,
                "display": "none",
                "height": 6,
                "left": 2,
                "position": "absolute",
                "right": 2,
              }
            }
          >
            <div
              className="uui-thumb-horizontal"
            />
          </div>
          <div
            style={
              Object {
                "borderRadius": 3,
                "bottom": 2,
                "display": "none",
                "position": "absolute",
                "right": 2,
                "top": 2,
                "width": 6,
              }
            }
          >
            <div
              className="uui-thumb-vertical"
            />
          </div>
        </div>
        <div
          className="uui-scroll-shadow-top"
        />
        <div
          className="uui-scroll-shadow-bottom"
        />
      </div>
    </div>
    <div
      className="root size-36 background-none padding-24 vPadding-24 spacing-12 modalFooter container align-items-center"
    >
      <div
        className="flexSpacer"
      />
      <div
        className="container uui-button-box uui-enabled -clickable root size-36 color-night500 fill-white style-square"
        href={null}
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="uui-caption line-height-18 font-size-14 v-padding-8"
        >
          Cancel
        </div>
      </div>
      <div
        className="container uui-button-box uui-enabled -clickable root size-36 color-grass fill-solid style-square"
        href={null}
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="uui-caption line-height-18 font-size-14 v-padding-8"
        >
          Select
        </div>
      </div>
    </div>
  </div>
</div>
`;
